src/game/client/neo/ui/neo_hud_friendly_marker.cpp

```diff
@@ -150,133 +191,260 @@ void CNEOHud_FriendlyMarker::DrawPlayerForTeam(C_Team *team, const C_NEO_Player
+	}
 
-			if (cl_neo_hud_iff_verbosity.GetInt() >= 1) 
-			{
-				int healthMode = cl_neo_hud_health_mode.GetInt();
-				V_snprintf(textASCII, MAX_MARKER_STRLEN, healthMode ? "%dhp" : "%d%%", player->GetDisplayedHealth(healthMode));
-				DisplayText(textASCII, drawOutline);
-			}
+	vgui::surface()->DrawSetTextColor(FadeColour(teamColor, fadeTextMultiplier));
 
-			static constexpr int HEALTHBAR_WIDTH = 64;
-			static constexpr int HEALTHBAR_HEIGHT = 6;
+	if (settings->bShowName) {
+		auto playerName = player->GetNeoPlayerName();
+		const char *playerClantag = player->GetNeoClantag();
 
-			if (cl_neo_hud_iff_healthbars.GetBool()) 
-			{
-				int healthBarWidth = Ceil2Int(HEALTHBAR_WIDTH * Min((float)player->GetHealth() / player->GetMaxHealth(), 1.0f)); // Clamp in case someone forgot to set max health, or it hasn't been sent by the server yet.
-				int healthBarYPos = y + (drawOutline ? 0 : m_iMarkerHeight) + textYOffset + 4;
-				surface()->DrawSetColor(FadeColour(teamColor, fadeTextMultiplier / 2));
-				surface()->DrawOutlinedRect(x - HEALTHBAR_WIDTH / 2, healthBarYPos, x + HEALTHBAR_WIDTH / 2, healthBarYPos + HEALTHBAR_HEIGHT);
-				surface()->DrawFilledRect(x - HEALTHBAR_WIDTH / 2, healthBarYPos, x - HEALTHBAR_WIDTH / 2 + healthBarWidth, healthBarYPos + HEALTHBAR_HEIGHT);
-			}
+		if (settings->bPrependClantagToName && playerClantag && playerClantag[0])
+		{
+			V_sprintf_safe(textASCII, "[%s] %s", playerClantag, playerName);
 		}
+		else
+		{
+			V_strcpy_safe(textASCII, playerName);
+		}
+		DisplayText(textASCII, y, settings->iMaxNameLength);
+	}
+}
+
+Color CNEOHud_FriendlyMarker::GetTeamColour(int team)
+{
+	return (team == TEAM_NSF) ? COLOR_NSF : COLOR_JINRAI;
+}
 
-		if (!drawOutline)
+union NeoIFFMarkerVariant
+{
+	int iVal;
+	float flVal;
+	bool bVal;
+};
+
+enum NeoIFFMarkerVariantType
+{
+	NEOIFFMARKERVARTYPE_INT = 0,
+	NEOIFFMARKERVARTYPE_FLOAT,
+	NEOIFFMARKERVARTYPE_BOOL,
+};
+
+static constexpr const NeoIFFMarkerVariantType NEOIFFMARKER_SEGMENT_VARTYPES[NEOIFFMARKER_SEGMENT__TOTAL] = {
+	NEOIFFMARKERVARTYPE_INT,   // NEOIFFMARKER_SEGMENT_I_INITIALOFFSET
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_SHOWDISTANCE
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_VERBOSEDISTANCE
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_SHOWSQUADMARKER
+	NEOIFFMARKERVARTYPE_FLOAT,   // NEOIFFMARKER_SEGMENT_FL_SQUADMARKERSCALE
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_SHOWHEALTHBAR
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_SHOWHEALTH
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_SHOWNAME
+	NEOIFFMARKERVARTYPE_BOOL,   // NEOIFFMARKER_SEGMENT_B_PREPENDCLANTAGTONAME
+	NEOIFFMARKERVARTYPE_INT,   // NEOIFFMARKER_SEGMENT_I_MAXNAMELENGTH
+};
+
+bool ImportMarker(FriendlyMarkerInfo *crh, const char *pszSequence)
+{
+	int iPrevSegment = 0;
+	int iSegmentIdx = 0;
+	NeoIFFMarkerVariant vars[NEOIFFMARKER_SEGMENT__TOTAL] = {};
+
+	const int iPszSequenceSize = V_strlen(pszSequence);
```

Comment:
Nit: would prefer `...Len` instead of `...Size` for the var name to avoid confusion